src/robot/libraries/Process.py

```diff
@@ -1075,18 +1093,32 @@ def _construct_env(self, env, extra):
 
     def _get_initial_env(self, env, extra):
         if env:
-            return {system_encode(k): system_encode(env[k]) for k in env}
+            result = {}
+            for name, value in env.items():
+                name = system_encode(name)
+                if isinstance(value, Secret):
+                    result[name] = system_encode(value.value)
+                    # remember secrets
+                    self.env_secret_keys.append(name)
```

Comment:
The comment is unnecessary. Perhaps `secret_env_keys` instead of `env_secret_keys`?